### PR TITLE
Commerce 11661

### PR DESCRIPTION
**Pull request by jaydawu**
https://liferay.atlassian.net/browse/COMMERCE-12323
Add automation and resent from https://github.com/liferay-commerce/liferay-portal/pull/4208

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:reopen

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 1]
```diff
@@ -401,6 +512,237 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-12326. If the SKU/UOM combination is set, the admin deletes a placed order, the inventory quantities from the SKU/UOM combination can be restocked correctly"
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:515]: If the SKU/UOM combination is set, when the admin deletes a placed order, can view the inventory quantities from the SKU/UOM combination be restocked correctly

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 2]
```diff
@@ -401,6 +512,237 @@ definition {
...
+	@priority = 5
+	test CanRestockInventorySKUUOMByDeletingTheOrder {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:517]: CanRestockSKUUOMInventoryAfterDeletingAnOrder

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 3]
```diff
@@ -401,6 +512,237 @@ definition {
...
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false${line.separator}feature.flag.COMMERCE-11287=true";
+		property portal.acceptance = "true";
+
+		task ("Given the buyer adds product ABS Sensor to the cart as quantity 2") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:521]: Given the buyer adds the product ABS Sensor to the cart in the quantity of 2

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 4]
```diff
@@ -401,6 +512,237 @@ definition {
...
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "test test");
+
+			CommerceInventory.openInventoryAdmin();
+
+			CommerceNavigator.searchEntry(entryName = "MIN93015");
+		}
+
+		task ("Then the quantities for UOM1 On Hand:140, Available:138, On Order:2, Incoming:0") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:566]: Then the quantities for UOM1 can be viewed as On Hand:140, Available:138, On Order:2, Incoming:0

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 5]
```diff
@@ -401,6 +512,237 @@ definition {
...
+					key_cellName1 = "MIN93015",
+					key_cellName2 = "uomKey1",
+					key_columnNumber = ${uomIndex},
+					locator1 = "CommerceEntry#TABLE_CELL_BY_COLUMN_ORDER_NO_TITLE",
+					value1 = ${uomQuantity});
+			}
+		}
+	}
+
+	@description = "This is a test for COMMERCE-12327. If the SKU/UOM combination is set, the admin deletes the shipment for a placed order, the inventory quantities from the SKU/UOM combination can be restocked correctly"
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:603]: If the SKU/UOM combination is set, when the admin deletes the shipment for a placed order, can view the inventory quantities from the SKU/UOM combination be restocked correctly

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 6]
```diff
@@ -401,6 +512,237 @@ definition {
...
+	@priority = 5
+	test CanRestockInventorySKUUOMByDeletingTheShipment {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:605]: CanRestockSKUUOMInventoryAfterDeletingAShipment

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 7]
```diff
@@ -401,6 +512,237 @@ definition {
...
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false${line.separator}feature.flag.COMMERCE-11287=true";
+		property portal.acceptance = "true";
+
+		task ("Given the buyer adds product ABS Sensor to the cart as quantity 2") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:609]: Given the buyer adds the product ABS Sensor to the cart in the quantity of 2

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 8]
```diff
@@ -401,6 +512,237 @@ definition {
...
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "test test");
+
+			CommerceInventory.openInventoryAdmin();
+
+			CommerceNavigator.searchEntry(entryName = "MIN93015");
+		}
+
+		task ("Then the quantities for UOM1 On Hand:140, Available:138, On Order:2, Incoming:0") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:654]: Then the quantities for UOM1 can be viewed as On Hand:140, Available:138, On Order:2, Incoming:0

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase [hunk 9]
```diff
@@ -401,6 +512,237 @@ definition {
...
+				AssertTextEquals(
+					key_cellName1 = "MIN93015",
+					key_cellName2 = "uomKey1",
+					key_columnNumber = ${uomIndex},
+					locator1 = "CommerceEntry#TABLE_CELL_BY_COLUMN_ORDER_NO_TITLE",
+					value1 = ${uomQuantity});
+			}
+		}
+
+		task ("But if the admin completes starting the checkout but doesn't ship the products") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceInventory.testcase:668]: But if the admin starts the shipment process of the order

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("When the admin searches for MIN93015 on the inventory admin page") {
+			CommerceInventory.openInventoryAdmin();
+
+			CommerceNavigator.searchEntry(entryName = "MIN93015");
+		}
+
+		task ("Then the new column UOM appears next to the SKU column to the right with an empty value") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:32]: "Then the new column for UOM can be viewed next to the SKU column with an empty value"

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+				key_cellName1 = "MIN93015",
+				key_cellName2 = "",
+				key_columnNumber = 2,
+				locator1 = "CommerceEntry#TABLE_CELL_BY_COLUMN_ORDER_NO_TITLE",
+				value1 = "");
+
+			Pagination.viewResults(results = "Showing 1 to 1 of 1 entries.");
+		}
+
+		task ("When 2 UOMs to SKU MIN93015 are created") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:47]: When 2 UOMs are added to the SKU MIN93015

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+			for (var index : list "1,2") {
+				CommerceJSONSKUsAPI._addCommerceUOM(
+					basePrice = ${index},
+					productSKU = "MIN93015",
+					uomKey = "uomKey${index}",
+					uomName = "uomName${index}");
+			}
+		}
+
+		task ("Then two separate line items for each SKU/UOM combination are searched") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:57]: Then two separate line items for each SKU/UOM combination can be viewed

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+				task ("And the SKU and UOM Key are shown in the header") {
+					AssertElementPresent(
+						key_title = "MIN93015 uomKey${uomIndex}",
+						locator1 = "Header#H3_TITLE");
+				}
+			}
+		}
+	}
+
+	@description = "This is a test for COMMERCE-12324. The admin can verify the inventory information from the SKU/UOM combination on the product SKU page"
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:127]: The admin can verify the inventory information of the SKU/UOM combination on the Product's SKU page.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+	@priority = 5
+	test VerifyUOMInventoryInfoOnSKUPage {
+		property portal.acceptance = "true";
+
+		task ("Given a Minium Site is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("When the admin goes into MIN93015's sku panel -> Inventory tab on the product admin page") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:136]: When the admin goes to the MIN93015's SKU panel -> Inventory tab on the product admin page

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceNavigator.gotoEntry(entryName = "MIN93015");
+
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Inventory");
+		}
+
+		task ("Then the new column UOM appears next to the warehouse column to the right with an empty value") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:150]: Then the new column of UOM can be viewed next to the warehouse column with an empty value

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+				value1 = "UOM");
+
+			CommerceSKUs.viewSkuUomWarehouse(
+				uomKey = "",
+				uomLists = "0:Italy,1:United States - Northeast,2:United States - Southwest");
+
+			CommerceSKUs.viewSkuUomInventory(uomQuantityLists = "0:20,1:60,2:60");
+		}
+
+		task ("When one UOM to SKU MIN93015 is created") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:163]: When one UOM is added to the SKU MIN93015

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+			CommerceEntry.gotoMenuTab(menuTab = "Inventory");
+
+			CommerceSKUs.viewSkuUomWarehouse(
+				uomKey = "uomKey1",
+				uomLists = "0:Italy,1:United States - Northeast,2:United States - Southwest");
+
+			CommerceSKUs.viewSkuUomInventory(uomQuantityLists = "0:20,1:60,2:60");
+		}
+
+		task ("When 2nd UOM to SKU MIN93015 is created") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:185]: When 2nd UOM are added to the SKU MIN93015

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+			CommerceJSONSKUsAPI._addCommerceUOM(
+				basePrice = 20,
+				productSKU = "MIN93015",
+				uomKey = "uomKey2",
+				uomName = "uomName2");
+		}
+
+		task ("Then the three warehouses are repeated for each SKU/UOM combination") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:193]: Then the warehouses are shown correctly for each SKU/UOM combination

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 19]
```diff
@@ -0,0 +1,238 @@
...
+			CommerceSKUs.viewSkuUomWarehouse(
+				uomKey = "uomKey1",
+				uomLists = "0:Italy,2:United States - Northeast,4:United States - Southwest");
+
+			CommerceSKUs.viewSkuUomWarehouse(
+				uomKey = "uomKey2",
+				uomLists = "1:Italy,3:United States - Northeast,5:United States - Southwest");
+		}
+
+		task ("And the 2nd SKU/UOM combination quantities are set to 0 as default") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:209]: And the 2nd SKU/UOM combination quantities are set to 0 by default

**Review by baharturk — CHANGES_REQUESTED** [review]
Hi, please take a look at my comments.

Thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
@baharturk Tests updated, please take a look, thanks!

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 20]
```diff
@@ -608,4 +608,225 @@ definition {
...
+						key_title = "MIN93015 uomKey${uomIndex}",
+						locator1 = "Header#H3_TITLE");
+				}
+			}
+		}
+	}
+
+	@description = "This is a test for COMMERCE-12324. The admin can verify the inventory information of the SKU/UOM combination on the Product's SKU page"
+	@priority = 5
+	test VerifyUOMInventoryInfoOnSKUPage {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:723]: Why did you omit Can in the beginning? We almost always start with Can, do you have a specific reason?

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 21]
```diff
@@ -608,4 +608,225 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-12324. The admin can verify the inventory information from the SKU/UOM combination on the Inventory admin page"
+	@priority = 5
+	test VerifyUOMInventoryInfoOnInventoryPage {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:613]: Why did you omit Can in the beginning? We almost always start with Can, do you have a specific reason?

**Comment by jaydawu** [comment]
ci:reopen

**Pull request closed without merging**